internal/services/conditionalaccess/conditional_access_policy_resource.go

```diff
@@ -172,18 +174,53 @@ func conditionalAccessPolicyResource() *schema.Resource {
 						},
 
 						"client_app_types": {
-							Type:     schema.TypeList,
-							Required: true,
+							Type:             schema.TypeList,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 								ValidateFunc: validation.StringInSlice([]string{
-									"all",
-									"browser",
-									"mobileAppsAndDesktopClients",
-									"exchangeActiveSync",
-									"easSupported",
-									"other",
-								}, false),
+									msgraph.ConditionalAccessClientAppTypeAll,
+									msgraph.ConditionalAccessClientAppTypeBrowser,
+									msgraph.ConditionalAccessClientAppTypeEasSupported,
+									msgraph.ConditionalAccessClientAppTypeExchangeActiveSync,
+									msgraph.ConditionalAccessClientAppTypeMobileAppsAndDesktopClients,
+									msgraph.ConditionalAccessClientAppTypeOther,
+								}, true),
+							},
+						},
+
+						"devices": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"device_filter": {
```

Comment:
given devices is just a block with a block inside, would it make sense to just move device_filter up a level and remove the devices block entirely?

Reply:
I went with the `device` block as there are a bunch of other properties in here that we'll add as soon as we have worked out togglable beta/preview support, or those properties make it to the stable API (whichever comes first) - but this will happen fairly soon so I didn't want to cause a deprecation a few weeks down the line.